shell/key-bindings.zsh

```diff
@@ -112,7 +112,7 @@ fzf-history-widget() {
   # history lines, is loaded, and that Perl is installed for multi-line output.
   if zmodload -F zsh/parameter p:history 2>/dev/null && (( ${#commands[perl]} )); then
     selected="$(printf '%1$s\t%2$s\000' "${(vk)history[@]}" |
-      perl -0 -ne 'if (!$seen{(/^\s*[0-9]+\**\s+(.*)/, $1)}++) { s/\n/\n\t/gm; print; }' |
+      perl -0 -ne 'if (!$seen{(/^\s*[0-9]+\**\s+((?s).*)/, $1)}++) { s/\n/\n\t/gm; print; }' |
```

Comment:
```suggestion
      perl -0 -ne 'if (!$seen{(/^\s*[0-9]+\**\t(.*)/s, $1)}++) { s/\n/\n\t/g; print; }' |
```

I agree with using `//s` and removing `//m`.

---

Instead of a `\s+`, replace it with a `\t`, which we know must follow a number at least once.

---

One might have also considered replacing `[0-9]` with `\d` to save another three characters, but after reading the `man` page, I would keep the current status quo.

```sh
man perlrecharclass | less --pattern "Digits"
       Digits

       "\d" matches a single character considered to be a decimal digit.  If
       the "/a" regular expression modifier is in effect, it matches [0-9].
       Otherwise, it matches anything that is matched by "\p{Digit}", which
       includes [0-9]. …

       What this means is that unless the "/a" modifier is in effect "\d" not
       only matches the digits '0' - '9', but also Arabic, Devanagari, and
       digits from other languages.  This may cause some confusion, and some
       security issues.

       Some digits that "\d" matches look like some of the [0-9] ones, but
       have different values.  For example, BENGALI DIGIT FOUR (U+09EA) looks
       very much like an ASCII DIGIT EIGHT (U+0038).  An application that is
       expecting only the ASCII digits might be misled, or if the match is
       "\d+", the matched string might contain a mixture of digits from
       different writing systems that look like they signify a number
       different than they actually do.  "num()" in Unicode::UCD can be used
       to safely calculate the value, returning "undef" if the input string
       contains such a mixture.

       What "\p{Digit}" means (and hence "\d" except under the "/a" modifier)
       is "\p{General_Category=Decimal_Number}", or synonymously,
       "\p{General_Category=Digit}".  Starting with Unicode version 4.1, this
       is the same set of characters matched by "\p{Numeric_Type=Decimal}".
       But Unicode also has a different property with a similar name,
       "\p{Numeric_Type=Digit}", which matches a completely different set of
       characters.  These characters are things such as "CIRCLED DIGIT ONE" or
       subscripts, or are from writing systems that lack all ten digits.
       …
```

Reply:
Sounds good to me.